app/root.tsx

```diff
@@ -102,6 +110,29 @@ function Document({ children, title }: PropsWithChildren<{ title?: string }>) {
 }
 
 function App() {
+  let transition = useNavigation();
```

Comment:
man this is good, but can you please put it in a custom hook, so we keep the Root component clean. Thanks